healvis/sky_model.py

```diff
@@ -168,3 +163,78 @@ def write_hdf5(self, filename):
                         dset = fileobj.create_dataset(k, data=d)
                     else:
                         dset = fileobj.create_dataset(k, data=d, compression='gzip', compression_opts=9)
+
+
+def flat_spectrum_noise_shell(sigma, freq_array, Nside, Nskies, ref_chan=0, shared_mem=False):
+    """
+    Make a flat-spectrum noise-like shell.
+
+    Args:
+        sigma : float
+            Power spectrum amplitude
+        freq_array : ndarray
+            Frequencies [Hz]
+        Nside : int
+            HEALpix Nside resolution
+        Nskies : int
+            Number of indepenent skies to simulate
+        ref_chan : int
+            freq_array reference channel index for comoving volume factor
+        shared_mem : bool
+            If True use mparray to generate data
+
+    Returns:
+        data : ndarray, shape (Nskies, Npix, Nfreqs)
+            EoR shell as HEALpix maps
+    """
+    # generate empty array
+    Nfreqs = len(freq_array)
+    Npix = hp.nside2npix(Nside)
+    if shared_mem:
+        data = mparray((Nskies, Npix, Nfreqs), dtype=float)
+    else:
+        data = np.zeros((Nskies, Npix, Nfreqs), dtype=float)
+
+    # setup parameters
+    dnu = np.diff(freq_array)[0]
+    om = 4 * np.pi / float(12 * Nside**2)
+    Zs = f21 / freq_array - 1.0
+    dV0 = comoving_voxel_volume(Zs[ref_chan], dnu, om)
+
+    # iterate over frequencies
+    for i in range(Nfreqs):
+        dV = comoving_voxel_volume(Zs[i], dnu, om)
+        amp = sigma * np.sqrt(dV0 / dV)
+        data[:, :, i] = np.random.normal(0.0, amp, (Nskies, Npix))
+
+    return data
+
+
+def gsm_shell(Nside, freq_array):
+    """
+    Generate a Global Sky Model shell
+
+    Args:
+        Nside : int
+            Nside resolution of HEALpix maps
+        freq_array : ndarray
+            Array of frequencies [Hz]
+
+    Returns:
+        data : ndarray, shape (Npix, Nfreqs)
+            GSM shell as HEALpix maps
+    """
+    assert pygsm_import, "Cannot use gsm_shell() when pygsm_import == False"
```

Comment:
Might want to rephrase... something like "The PyGSM module is needed to use the Global Sky Model"